src/eventer/eventer_ports_impl.c

```diff
@@ -223,14 +223,31 @@ eventer_ports_impl_trigger(eventer_t e, int mask) {
   const char *cbname;
   struct timeval __now;
   int fd, newmask;
+  int cross_thread = mask & EVENTER_CROSS_THREAD_TRIGGER;
 
   mask = mask & ~(EVENTER_RESERVED);
   fd = e->fd;
-  if(e != master_fds[fd].e) return;
+  if(cross_thread) {
+    if(master_fds[fd].e != NULL) {
+      mtevL(eventer_deb, "Attempting to trigger already-registered event fd: %d cross thread.\n", fd);
+    }
+    /* assert(master_fds[fd].e == NULL); */
+  }
   if(!pthread_equal(pthread_self(), e->thr_owner)) {
+    /* If we're triggering across threads, it can't be registered yet */
+    if(master_fds[fd].e != NULL) {
+      mtevL(eventer_deb, "Attempting to trigger already-registered event fd: %d cross thread.\n", fd);
+    }
```

Comment:
not if we want it to behave like it used to... that should become an assert at some point later, but now, we should carry on and risk it.